### ISSUE TITLE
[Feature Request]: HomeKit Secure Video

### ISSUE THREAD

**Issue by thecem** [Brandawg93/homebridge-nest-cam#464]
### Enhancement of issue

Homebridge support with Version 1.4.0 HomeKit Secure Video.
Actually scrypted with nest plugin is already supporting HomeKit Secure Video. So I think most users would love this!!!
And the best, it is much faster than the original app, with a lot of interruptions (even on symmetrical 1G fiber Internet connection with full wlan connection)…
Thanks a lot!

### Solution

Implement the nest cam plugin into the homebridge HomeKit Secure Video functionality.

**Comment by jricey** [comment]
Would very much love to get this working. Hoping it can come to be able to use the Google Nest Battery Doorbell with HSV

**Comment by Pottebaum** [comment]
I second, or third, this request, after seeing how well HKSV works with standard RTSP streams it would be amazing to be able to do the same from the Nest Hello camera.  Thanks!

**Comment by andrewfblack** [comment]
I would also like this added thanks

**Comment by djtaube** [comment]
I am hopeful that HKSV can be incorporated. This would allow me to keep the hardware I have instead of getting a different camera that does RTSP.

**Comment by n0rt0nthec4t** [comment]
I developed by own HomeKit accessory for my Nest devices (using HAP-NodeJS). Mainly as I like the challenge, don't use Homebridge and my Nest devices haven't been migrated to Google accounts. My Nest camera streaming was basically using the code that Brandon developed here.

I've started some initial code testing to see how HKSV would work with my Nest Hello and Nest Outdoor cam.. Attached is screenshot so far (mined the bluring in image.. :-). Still some ways togo and working through some issues found with HAP-NodeJS and HKSV

Ultimately not sure about it, mainly due to bandwidth requirements ie: Cam is uploading to Nest, while HomeKit is pulling down a stream, which in turn is being upload to iCloud.. 

<img width="809" alt="HKSV test" src="https://user-images.githubusercontent.com/19759130/161003036-a9e51d08-e3e5-4fed-8d06-d7c223e485a1.png">
V

**Comment by Brandawg93** [comment]
Exactly. The network issues present a real challenge for HSV. That's why I haven't published anything yet with HSV support because I know the bugs will come flooding in 😀

**Comment by n0rt0nthec4t** [comment]
Yes, definitely see issues with network and time it takes to get started on streams

**Comment by n0rt0nthec4t** [comment]
Been tinkering alot more and  have refined how this works. Yes, made it pre-buffer the video stream from a camera, so its basically; constantly recording from Nest. Bit more playing todo and happy to release the update to my repository for people to see

**Comment by Brandawg93** [comment]
Feel free to submit a PR here! I may release it under a beta tag for people to try for a while.

**Comment by krehman** [comment]
I would be interested in trying this out. Please let me know if it is released as a beta.

**Comment by Pottebaum** [comment]
That's great news you were able to make some progress on this! It is amazing how much better my other cameras that support HSV work.  Be interested in testing this out as well.

**Comment by n0rt0nthec4t** [comment]
As stated, I don't use HomeBridge or this plugin directly. I've uploaded my current development work on this to my repository. If someone wants to start converting Into typescript and make a branch for this plugin, go for it :-)

https://github.com/n0rt0nthec4t/HAP-NodeJS-Accessories

**Comment by n0rt0nthec4t** [comment]
With help from @krehman , we've refined the working code and also made a docker version if people want to try it out. @Brandawg93 do you have an email contact we can chat off-line if you'd be interested in a beta channel on here (back porting my code into this) and some other refinements 

Anyone else, like to test, move to ist own repository 
https://github.com/n0rt0nthec4t/Nest_accfactory

**Comment by Brandawg93** [comment]
Sure. Feel free to contact me at brandawg93@proton.me.

**Comment by GillesWi** [comment]
Hi! Any updates on this? What's the easiest way to install it on Homebridge, does it still work?

**Comment by GillesWi** [comment]
:)?
